src/main/java/org/kaazing/netx/ws/WebSocket.java

```diff
@@ -132,20 +132,11 @@
     public abstract InputStream getInputStream() throws IOException;
 
     /**
-     * Returns a {@link MessageReader} that can be used to receive <b>binary</b> and/or <b>text</b> messages.
-     * <p>
-     * @return WebSocketMessageReader   to receive binary and text messages
-     * @throws IOException if an I/O error occurs when connecting or connection is closed
-     */
-    public abstract MessageReader getMessageReader() throws IOException;
-
-    /**
-     * Returns a {@link MessageWriter} that can be used to send <b>binary</b> and/or <b>text</b> messages.
+     * Returns the maximum payload length that this connection will support. The default maximum payload length is 8192 bytes.
      *
-     * @return WebSocketMessageWriter   to send binary and/or text messages
-     * @throws IOException if an I/O error occurs when connecting or connection is closed
+     * @return maximum payload length for the connection
      */
-    public abstract MessageWriter getMessageWriter() throws IOException;
+    public abstract int getMaxPayloadLength();
```

Comment:
Is this intended to convey maximum WebSocket message size, or maximum WebSocket frame size?